### DISCUSSION TITLE
Error: Failure to solve real system of equations

### DISCUSSION THREAD

**Discussion by MohammadBabaeifar** [OpenModelica/OpenModelica#10890] [Q&A]
Hello,
I have created a model which contains a three-phase 6-pulse bridge. When I replace a current sensor on the dc leg of the bridge with a wire (which is theoretically equivalent), and run the simulation in Dymola, I receive this error: **Failure to solve real system of equations** followed by these messages:
**the system is singular, which may be caused by redundant initial conditions.**
**Setting Advanced.Translation.SimplifyAfterIndexReduction=true may select better default initial conditions that avoid this**

In OpenModelica however I don not receive such errors.

So, in  Dymola in the Simulation>>setup>>translation: I activated this: **Evaluate parameters to reduce models.**
The simulation then works with the same results as before removing the sensor.

Though, I was wondering if you could help me in finding the reason for the above-mentioned error and messages and how to solve it.

**Comment by casella** [comment]
@MohammadBabaeifar, 3-phase circuits with inductors are high-index systems: in principle, you have three state variables for the three currents (corresponding to terms like der(i[1]), der(i[2]), and der(i[3]), but then if you connect them in a start or delta fashion, constraints such as i1 + i2 + i3 are added to the problem, which directly constrain the state variables. Index reduction and dummy derivatives are applied, eventually kicking out one state variable in the process and only keeping two of them.

If the original components (e.g. inductors) had initial equations for their currents, when the index reduction is performed one of the initial equations becomes redundant. Modelica tools have heuristic algorithms to figure out redundant _but consistent_ initial equations in these cases, but they don't always work. The problem is that the variables i1, i2, and i3 are normally not the same that show up under derivative sign, and figuring out symbolically that they are aliases is non trivial.

You can try to disable the initial equation on one of the inductors to see if the problem is solved.

**Comment by MohammadBabaeifar** [comment]
Dear Professor Casella, 
I am sorry that I forgot to reply on your comment and inform you that my problem was solved.

In fact, the developers in Calytex suggested an approach like what you mentioned. I actually disabled the initial value in some inductors by setting their initial current to fixed=false. The problem was solved then.
Thanks for your help.
